enhancements/implicit-catalog-versioning.md

```diff
@@ -150,3 +150,182 @@ There was some significant discussion around bumping the CSV API version to `v1a
 Simply making the change outright given that the CSV API is still in an `alpha` version. We chose not to go this route given there are already many production workloads in the wild using this API.
 
 Attempting to only make this requirement exist in the case where new catalog images built from `opm` are present, in which case anything could be migrated by looking up the relevant subscription during the migration to `v1alpha2` by querying the API for the new catalog image. For CSVs not tied to the subscription, just remove the `replaces` field and choose a sane default for the version. We chose not to go this route because there are simply too many moving pieces and it significantly increased the scope of this enhancement and tied it too closely with a similar migration effort around replacing app-registry support with native container images.
+
+### Tooling
+
+In order to actually reason about and interact with existing indexes, we need a convenient method of inspecting them. Currently, the only way to determine what is actually *in* the index is to fetch the image, run it, and query the served grpc API. Even then, you need a somewhat involved understanding of the replacement chain abstraction to follow the graph down for a given channel. Instead, we will create a new `opm` command `inspect` that abstracts the need for someone to serve the database to determine what packages and versions are in the registry. For example:
```

Comment:
I have a question about how this works. `opm inspect` sounds great. Does it actually query the DB via the gRPC api under the hood, or will it parse the bundle DB in some other way?

Reply:
Serving and then querying the grpc api would be relatively high overhead. Instead, we will just do what most other `opm` commands do which is just create a direct connection to the sql db and query it.